# 🏥 HealthConnect Appointment Analysis

## Week 7: Testing, Refinement and End-to-End Validation

**AnalystLab Africa — HealthConnect Experience Lab**

**Data Analytics Track**

Prepared by **Hudu Yusuf Ibrahim**

---

## Week 6 → Week 7 Transition

**1. Main Week 6 output:**  
An Advanced Analytics & Decision Support notebook with statistical validation using Chi-square tests and Cramér's V, a refined KPI structure including the combined Lead Time × Previous No-Show analysis, an improved Power BI dashboard, and completed cross-track integration with two Data Science interns.

**2. Most important component/result:**  
The statistically validated **Lead Time × Previous No-Show** segment was the strongest observed association in the analysis, with a **Cramér's V of 0.189**. This finding was independently corroborated by the Data Science modelling results and therefore became a key focus for further testing.

**3. Main issue, limitation or dependency identified:**  
Two issues were identified during Week 6. First, a difference was observed between the Data Analytics and Data Science calculations for the same segment (**64.1% vs 70.5%**), which was traced to a difference in denominator definition and resolved through discussion, but was not yet re-verified directly within the Power BI dashboard. Second, the **Lead Time Band** sort-order issue in Power BI remained unresolved.

**4. Component requiring testing:**  
The primary component requiring testing is the **Power BI dashboard**, particularly whether its displayed KPI values match independent calculations from the underlying dataset and whether its filters, slicers, and visual calculations behave as intended.

**5. Relevant track(s) for Week 7:**  
Week 7 will primarily involve **Data Analytics self-validation** of the dashboard and analytical outputs. The Data Science connection will also be revisited where necessary to ensure that the 64.1%/70.5% discrepancy is clearly documented and does not create confusion between the two teams' materials.

**6. What will be tested:**  
A systematic accuracy audit will be performed across the dashboard, including the **6 KPI cards, comparison charts, and 2 slicers**. Each important displayed value will be compared against an independent recalculation from the raw dataset. Dashboard filtering, KPI calculations, chart values, the combined segment definition, and the Lead Time Band ordering will also be tested.

**7. What is expected to improve or be validated:**  
The testing will determine whether the dashboard is fully accurate and trustworthy, identify and correct any discrepancies found, and improve dashboard usability where necessary. The **Lead Time Band sort-order issue** will also be resolved where possible or formally documented if a technical limitation remains. The final objective is to ensure that the validated analytical findings and dashboard are reliable and ready for the next stage of HealthConnect integration.

---

## Week 7 Testing Readiness

### Testing Objective

The main objective of Week 7 testing is to verify that the important analytical findings and Power BI dashboard outputs from Week 6 are accurate, consistent, and reliable.

The testing will compare dashboard results against independent calculations from the underlying HealthConnect dataset and will also assess whether dashboard filters, KPI calculations, visual values, and segment definitions behave as intended.

### Component Being Tested

The primary component for testing is the **HealthConnect Power BI dashboard**, together with the analytical calculations supporting its KPIs and visuals.

The testing will focus on:

- KPI card values
- No-Show Rate calculations
- Lead Time analysis
- Previous No-Show analysis
- Combined Lead Time × Previous No-Show segment
- Other comparison charts
- Appointment Type slicer
- Gender slicer
- Lead Time Band ordering
- Consistency between dashboard values and independent dataset calculations

### Testing Approach

The testing will be performed in stages:

1. **KPI validation**  
   Independently recalculate the dashboard KPI values from the underlying dataset and compare them with the values displayed in Power BI.

2. **Visual validation**  
   Compare important chart values against independent calculations from the dataset.

3. **Segment validation**  
   Recalculate important No-Show segments, particularly the **Lead Time × Previous No-Show** segment, and confirm that the dashboard uses the intended denominator and segment definitions.

4. **Filter and slicer testing**  
   Test the Appointment Type and Gender slicers to confirm that the relevant KPI cards and visuals respond correctly.

5. **Calculation testing**  
   Check that the DAX measures produce the expected results and that filtering does not produce unexpected or inconsistent values.

6. **Sort-order testing**  
   Test the Lead Time Band ordering to determine whether the categories appear in the intended sequence:
   - Same week (0–7 days)
   - 2–3 weeks (8–21 days)
   - 3+ weeks (22–60 days)

7. **Retesting after refinement**  
   Where an issue is identified, the dashboard or calculation will be corrected and tested again to confirm that the refinement resolved the issue.

### Testing Success Criteria

The dashboard will be considered ready when:

- KPI values match independent calculations.
- Important chart values match the underlying dataset.
- The Lead Time × Previous No-Show segment uses a clearly defined and consistent denominator.
- Slicers filter the intended dashboard components correctly.
- DAX calculations return expected results.
- Lead Time Band categories appear in the correct order.
- Any identified discrepancies are corrected and successfully retested.
- Remaining limitations are clearly documented rather than hidden.

### Testing Evidence to Capture

Evidence will include:

- Independent calculation outputs from Python.
- Screenshots of relevant Power BI visuals and KPI cards.
- Slicer testing results.
- Before-and-after screenshots where a refinement is made.
- Testing records showing expected results, actual results, pass/fail status, actions taken, and retest results.

No issue will be assumed in advance. If a component passes testing, the successful validation will be documented. If an issue is identified, the issue will be investigated, corrected where possible, and retested.

---

## Step 3: KPI Validation

The five KPI cards displayed on the Week 6 Power BI dashboard were independently recalculated using the raw HealthConnect appointment dataset.

The recalculations were performed in this Week 7 testing notebook using Python and pandas. The calculations followed the same underlying definitions used for the Power BI measures, allowing the dashboard values to be compared directly with independently calculated results.

The validation focused on confirming that the correct numerator, denominator, and segment definitions were being used for each KPI.

In [1]:
import pandas as pd
df = pd.read_csv("HealthConnect_Appointment_Data.csv")

# Independent KPI validation

# 1. Overall No-Show Rate
overall_no_show_rate = (
    df["appointment_outcome"].eq("No-Show").mean()
)

# 2. No-Show Rate for appointments with no reminder
no_reminder = df[df["reminder_sent"] == "No"]

no_reminder_rate = (
    no_reminder["appointment_outcome"].eq("No-Show").mean()
)

# 3. No-Show Rate for appointments with previous No-Shows
previous_no_show = df[df["previous_no_shows"] > 0]

previous_no_show_rate = (
    previous_no_show["appointment_outcome"].eq("No-Show").mean()
)

# 4. No-Show Rate for 3+ week lead time
three_plus_weeks = df[df["booking_lead_days"] >= 22]

three_plus_week_rate = (
    three_plus_weeks["appointment_outcome"].eq("No-Show").mean()
)

# 5. No-Show Rate for 3+ weeks + Previous No-Show
highest_observed_segment = df[
    (df["booking_lead_days"] >= 22) &
    (df["previous_no_shows"] > 0)
]

highest_segment_rate = (
    highest_observed_segment["appointment_outcome"].eq("No-Show").mean()
)

# Display results
print(f"Overall No-Show Rate: {overall_no_show_rate:.2%}")
print(f"No Reminder: {no_reminder_rate:.2%}")
print(f"Previous No-Show: {previous_no_show_rate:.2%}")
print(f"3+ Week Lead Time: {three_plus_week_rate:.2%}")
print(f"Highest Observed Segment: {highest_segment_rate:.2%}")



Overall No-Show Rate: 48.46%
No Reminder: 51.39%
Previous No-Show: 55.41%
3+ Week Lead Time: 56.94%
Highest Observed Segment: 64.09%


### Observation

The independently calculated KPI values matched the Power BI dashboard values after rounding to one decimal place.

The validation confirmed that:

- The Overall No-Show Rate is **48.46%**, displayed as **48.5%** in Power BI.
- The No Reminder rate is **51.39%**, displayed as **51.4%**.
- The Previous No-Show rate is **55.41%**, displayed as **55.4%**.
- The 3+ Week Lead Time rate is **56.94%**, displayed as **56.9%**.
- The Highest Observed Segment rate is **64.09%**, displayed as **64.1%**.

No discrepancy was identified in the five KPI calculations.

The same denominator definition used in the Data Analytics analysis was applied, meaning cancelled appointments remain included in the denominator.

### KPI Validation Results

| KPI | Power BI Value | Independent Value | Result |
|---|---:|---:|---|
| Overall No-Show Rate | 48.5% | 48.46% | **PASS** |
| No Reminder | 51.4% | 51.39% | **PASS** |
| Previous No-Show | 55.4% | 55.41% | **PASS** |
| 3+ Week Lead Time | 56.9% | 56.94% | **PASS** |
| Highest Observed Segment | 64.1% | 64.09% | **PASS** |

---

## Step 4: Analytical Validation

The key analytical findings from Weeks 5 and 6 were independently validated using the raw HealthConnect appointment dataset.

The purpose of this testing was to confirm that the main No-Show patterns identified during the earlier analysis remained consistent when recalculated independently.

The validation focused on:

- Booking Lead Time and No-Show Rate
- Previous No-Show History and No-Show Rate
- Combined Lead Time × Previous No-Show analysis
- Chi-square tests and Cramér's V used to validate the main associations

The results were compared with the Week 5 and Week 6 findings to identify any inconsistencies or calculation errors.

In [2]:
# Create the same lead-time bands used in the Week 5/6 analysis

df["lead_time_band"] = pd.cut(
    df["booking_lead_days"],
    bins=[-1, 7, 21, 60],
    labels=[
        "Same week (0-7 days)",
        "2-3 weeks (8-21 days)",
        "3+ weeks (22-60 days)"
    ]
)

# Calculate No-Show Rate by lead-time band

lead_time_validation = (
    df.groupby("lead_time_band", observed=False)["appointment_outcome"]
      .apply(lambda x: (x == "No-Show").mean() * 100)
      .round(2)
)

lead_time_validation

lead_time_band
Same week (0-7 days)     27.81
2-3 weeks (8-21 days)    37.21
3+ weeks (22-60 days)    56.94
Name: appointment_outcome, dtype: float64

### Observation

The independently recalculated No-Show rates were **27.81%**, **37.21%**, and **56.94%** for Same Week, 2–3 Weeks, and 3+ Weeks respectively.

These results match the Week 5/6 findings after rounding.

The same pattern remains visible: the observed No-Show rate increases as booking lead time increases.

**Validation result: PASS**

In [3]:
# Create previous No-Show groups
import numpy as np

df["previous_no_show_group"] = np.where(
    df["previous_no_shows"] > 0,
    "Previous No-Show",
    "No Previous No-Show"
)

# Calculate No-Show Rate by previous No-Show history

previous_no_show_validation = (
    df.groupby("previous_no_show_group")["appointment_outcome"]
      .apply(lambda x: (x == "No-Show").mean() * 100)
      .round(2)
)

previous_no_show_validation

previous_no_show_group
No Previous No-Show    43.51
Previous No-Show       55.41
Name: appointment_outcome, dtype: float64

### Observation

The independently recalculated No-Show rates were **43.51%** for appointments with no previous No-Show history and **55.41%** for appointments with previous No-Show history.

These results match the Week 5/6 findings after rounding.

The higher observed No-Show rate among appointments with previous No-Show history therefore remains present in the raw dataset.

**Validation result: PASS**

In [4]:
# Combine lead time and previous No-Show history

df["lead_time_previous_segment"] = (
    df["lead_time_band"].astype(str)
    + " + "
    + df["previous_no_show_group"]
)

# Calculate No-Show Rate for each combined segment

combined_validation = (
    df.groupby("lead_time_previous_segment", observed=False)["appointment_outcome"]
      .apply(lambda x: (x == "No-Show").mean() * 100)
      .round(2)
      .sort_values()
)

combined_validation

lead_time_previous_segment
Same week (0-7 days) + No Previous No-Show     21.77
2-3 weeks (8-21 days) + No Previous No-Show    34.11
Same week (0-7 days) + Previous No-Show        36.19
2-3 weeks (8-21 days) + Previous No-Show       41.93
3+ weeks (22-60 days) + No Previous No-Show    51.70
3+ weeks (22-60 days) + Previous No-Show       64.09
Name: appointment_outcome, dtype: float64

### Observation

The combined Lead Time × Previous No-Show analysis reproduced the Week 5/6 pattern.

The lowest observed No-Show rate was **21.77%** for appointments with Same Week lead time and no previous No-Show history.

The highest observed No-Show rate was **64.09%** for appointments with **3+ weeks lead time and previous No-Show history**.

The dashboard displays this highest observed segment as **64.1%**, which is consistent with the independent calculation after rounding.

**Validation result: PASS**

In [5]:
from scipy.stats import chi2_contingency
import numpy as np

# Booking Lead Time vs Appointment Outcome

lead_time_table = pd.crosstab(
    df["lead_time_band"],
    df["appointment_outcome"]
)

chi2_lead, p_lead, dof_lead, expected_lead = chi2_contingency(
    lead_time_table
)

n_lead = lead_time_table.values.sum()
rows_lead, cols_lead = lead_time_table.shape

cramers_v_lead = np.sqrt(
    chi2_lead / (n_lead * min(rows_lead - 1, cols_lead - 1))
)

print("Booking Lead Time")
print("Chi-square:", round(chi2_lead, 2))
print("p-value:", p_lead)
print("Degrees of freedom:", dof_lead)
print("Cramér's V:", round(cramers_v_lead, 3))

Booking Lead Time
Chi-square: 280.09
p-value: 2.1299647664781796e-59
Degrees of freedom: 4
Cramér's V: 0.167


In [6]:
# Previous No-Show History vs Appointment Outcome

previous_table = pd.crosstab(
    df["previous_no_show_group"],
    df["appointment_outcome"]
)

chi2_previous, p_previous, dof_previous, expected_previous = chi2_contingency(
    previous_table
)

n_previous = previous_table.values.sum()
rows_previous, cols_previous = previous_table.shape

cramers_v_previous = np.sqrt(
    chi2_previous / (n_previous * min(rows_previous - 1, cols_previous - 1))
)

print("Previous No-Show History")
print("Chi-square:", round(chi2_previous, 2))
print("p-value:", p_previous)
print("Degrees of freedom:", dof_previous)
print("Cramér's V:", round(cramers_v_previous, 3))

Previous No-Show History
Chi-square: 69.86
p-value: 6.773628974738036e-16
Degrees of freedom: 2
Cramér's V: 0.118


In [7]:
# Combined Lead Time × Previous No-Show vs Appointment Outcome

combined_table = pd.crosstab(
    df["lead_time_previous_segment"],
    df["appointment_outcome"]
)

chi2_combined, p_combined, dof_combined, expected_combined = chi2_contingency(
    combined_table
)

n_combined = combined_table.values.sum()
rows_combined, cols_combined = combined_table.shape

cramers_v_combined = np.sqrt(
    chi2_combined / (n_combined * min(rows_combined - 1, cols_combined - 1))
)

print("Combined Lead Time × Previous No-Show")
print("Chi-square:", round(chi2_combined, 2))
print("p-value:", p_combined)
print("Degrees of freedom:", dof_combined)
print("Cramér's V:", round(cramers_v_combined, 3))

Combined Lead Time × Previous No-Show
Chi-square: 358.52
p-value: 6.198820527607995e-71
Degrees of freedom: 10
Cramér's V: 0.189


### Analytical Validation Conclusion

The Week 5 and Week 6 analytical findings were successfully reproduced using independent calculations from the raw dataset.

The validation confirmed that:

- Booking Lead Time remains associated with appointment outcome (χ² = **280.09**, Cramér's V = **0.167**).
- Previous No-Show History remains associated with appointment outcome (χ² = **69.86**, Cramér's V = **0.118**).
- The combined Lead Time × Previous No-Show analysis produced the highest Cramér's V among these tested analyses (χ² = **358.52**, Cramér's V = **0.189**).
- The highest observed segment remains **3+ weeks + Previous No-Show**, with a No-Show Rate of **64.09%**, displayed as **64.1%** in Power BI.

No analytical discrepancy was identified during this validation.

These results confirm that the main Week 6 findings are reproducible from the underlying dataset. The statistical results indicate association, not causation, and the Cramér's V values should not be interpreted as predictive-model performance.

**Degrees of freedom note:** The combined Lead Time × Previous No-Show test has **10 degrees of freedom** because the combined variable contains **6 categories** and appointment outcome contains **3 categories**. Therefore, `(6 − 1) × (3 − 1) = 10`. This differs from the Lead Time test (4 df) and Previous No-Show test (2 df) because those variables have fewer categories.

---

## Step 5: Segment Testing

The main Week 6 findings were further tested across relevant appointment and patient segments to determine whether the observed patterns remained consistent.

The testing focused on the relationship between Booking Lead Time and No-Show Rate across:

- Appointment Type
- Distance to Clinic
- Reminder Status

The combined Lead Time × Previous No-Show finding was also reviewed across Appointment Type.

The purpose of this testing was not to claim that the relationships are independent of other factors, but to determine whether the main observed patterns remained visible across different segments.

In [8]:
# No-Show Rate by Appointment Type and Lead Time

lead_time_by_appointment = pd.crosstab(
    df["appointment_type"],
    df["lead_time_band"],
    values=df["appointment_outcome"].eq("No-Show"),
    aggfunc="mean"
) * 100

lead_time_by_appointment.round(2)

lead_time_band,Same week (0-7 days),2-3 weeks (8-21 days),3+ weeks (22-60 days)
appointment_type,,,
Diagnostic Test,29.69,32.35,59.03
Follow-up,29.41,40.38,59.89
General Consultation,24.60,37.92,54.36
Specialist Consultation,30.65,34.07,56.73


### Observation: Appointment Type

The No-Show Rate increased as Booking Lead Time increased across all four appointment types.

The No-Show Rate increased from the Same Week category to the 3+ Week category for:

- Diagnostic Test
- Follow-up
- General Consultation
- Specialist Consultation

The increase ranged from approximately **26.1 to 30.5 percentage points** across the four appointment types.

This confirms that the main Lead Time pattern identified in Week 6 remains visible across appointment types.

**Validation result: PASS**

In [9]:
# No-Show Rate by Distance Band and Lead Time
df["distance_band"] = pd.cut(
    df["distance_to_clinic_km"],
    bins=[0, 5, 15, 45],
    labels=["<5km", "5-15km", "15km+"]
)

lead_time_by_distance = pd.crosstab(
    df["distance_band"],
    df["lead_time_band"],
    values=df["appointment_outcome"].eq("No-Show"),
    aggfunc="mean"
) * 100

lead_time_by_distance.round(2)

lead_time_band,Same week (0-7 days),2-3 weeks (8-21 days),3+ weeks (22-60 days)
distance_band,,,
<5km,24.05,29.45,57.81
5-15km,26.10,37.77,54.94
15km+,35.94,44.69,61.81


### Observation: Distance to Clinic

The No-Show Rate increased with longer Booking Lead Time across all three distance groups.

The 3+ Week category had the highest observed No-Show Rate within each distance group.

The increase from Same Week to 3+ Weeks ranged from approximately **25.9 to 33.7 percentage points**.

The results therefore support the Week 6 observation that the Lead Time pattern remains visible across different distance groups.

**Validation result: PASS**

In [10]:
# No-Show Rate by Reminder Status and Lead Time

lead_time_by_reminder = pd.crosstab(
    df["reminder_sent"],
    df["lead_time_band"],
    values=df["appointment_outcome"].eq("No-Show"),
    aggfunc="mean"
) * 100

lead_time_by_reminder.round(2)

lead_time_band,Same week (0-7 days),2-3 weeks (8-21 days),3+ weeks (22-60 days)
reminder_sent,,,
No,29.94,40.84,59.58
Yes,27.06,35.82,55.94


### Observation: Reminder Status

The No-Show Rate increased with longer Booking Lead Time in both reminder-status groups.

For appointments where no reminder was sent, the rate increased from **29.9%** in the Same Week group to **59.6%** in the 3+ Week group.

For appointments where a reminder was sent, the rate increased from **27.1%** to **55.9%**.

The Lead Time pattern therefore remained visible in both reminder-status groups.

This does not establish that reminder status has no effect or that the two factors are independent. It only confirms that the observed Lead Time pattern remains present within both reminder-status groups.

**Validation result: PASS**

In [11]:
# No-Show Rate by Appointment Type,
# Lead Time and Previous No-Show History

combined_by_appointment = pd.crosstab(
    df["appointment_type"],
    [df["lead_time_band"], df["previous_no_show_group"]],
    values=df["appointment_outcome"].eq("No-Show"),
    aggfunc="mean"
) * 100

combined_by_appointment.round(2)

lead_time_band          Same week (0-7 days)                   \
previous_no_show_group   No Previous No-Show Previous No-Show   
appointment_type                                                
Diagnostic Test                        16.67            46.43   
Follow-up                              23.68            36.67   
General Consultation                   18.62            33.01   
Specialist Consultation                27.27            36.17   

lead_time_band          2-3 weeks (8-21 days)                   \
previous_no_show_group    No Previous No-Show Previous No-Show   
appointment_type                                                 
Diagnostic Test                         35.90            27.59   
Follow-up                               36.32            46.72   
General Consultation                    33.44            44.76   
Specialist Consultation                 31.65            37.93   

lead_time_band          3+ weeks (22-60 days)                   
previous_no_show_group    No Previous No-Show Previous No-Show  
appointment_type                                                
Diagnostic Test                         52.23            68.05  
Follow-up                               56.36            64.81  
General Consultation                    47.66            63.28  
Specialist Consultation                 53.09            61.95

### Observation: Combined Lead Time × Previous No-Show

The combined analysis was reviewed across all four appointment types.

The **3+ weeks + Previous No-Show** segment had the highest observed No-Show Rate within each appointment type:

- Diagnostic Test: **68.0%**
- Follow-up: **64.8%**
- General Consultation: **63.3%**
- Specialist Consultation: **61.9%**

The rates ranged from **61.9% to 68.0%**, showing that the high observed No-Show pattern was present across all four appointment types.

One smaller subgroup showed a reversal in the 2–3 week category for Diagnostic Tests. This subgroup should be interpreted cautiously because of its smaller sample size.

Overall, the highest-risk observed segment identified in the broader analysis remained visible across appointment types.

**Validation result: PASS**

### Segment Testing Conclusion

The Week 6 analytical findings were tested across Appointment Type, Distance to Clinic, and Reminder Status.

The testing confirmed that:

- The No-Show Rate increased with longer Booking Lead Time across all appointment types.
- The Lead Time pattern remained visible across distance groups.
- The Lead Time pattern remained visible across both reminder-status groups.
- The combined **3+ weeks + Previous No-Show** segment remained the highest observed segment across all four appointment types.

These results provide additional validation that the main patterns identified in Week 6 are reproducible across relevant segments.

The findings remain descriptive and should not be interpreted as proof of causation, independence, or predictive performance.

---

## Step 6: Power BI Dashboard Testing

The Week 6 Power BI dashboard was tested to confirm that the displayed values, visuals, filters, and calculations behave as intended.

The testing focused on:

- KPI card values
- Comparison chart values
- Appointment Type slicer
- Gender slicer
- Lead Time Band ordering
- Combined Lead Time × Previous No-Show analysis
- Consistency between dashboard results and independently validated calculations

The dashboard was tested using different filter selections and compared against the expected results from the underlying HealthConnect dataset.

### KPI Card Display Test

The five KPI cards were checked against the independently validated values from Step 3.

The dashboard displayed:

- Overall No-Show Rate: **48.5%**
- No Reminder: **51.4%**
- Previous No-Show: **55.4%**
- 3+ Week Lead Time: **56.9%**
- Highest Observed Segment: **64.1%**

The displayed values matched the independently calculated results.

**Test result: PASS**

### Appointment Type Slicer Test

The Appointment Type slicer was tested using different appointment-type selections.

The dashboard visuals and KPI calculations responded to the selected appointment type, and the results changed appropriately when the selection was changed and cleared.

No unexpected filtering behaviour was observed.

**Test result: PASS**

### Gender Slicer Test

The Gender slicer was tested using different gender selections and then cleared.

The dashboard visuals and KPI calculations responded to the selected gender as expected.

No unexpected filtering behaviour was observed.

**Test result: PASS**

### Lead Time Band Ordering Test

The Lead Time Band ordering was tested to confirm that the categories are arranged according to their logical booking-time sequence.

A dedicated numeric sort column, `Lead Time Band Sort`, was created using `booking_lead_days` and applied to the `Lead Time Band` column.

The expected order was:

1. Same week (0–7 days)
2. 2–3 weeks (8–21 days)
3. 3+ weeks (22–60 days)

The dashboard was checked after applying the sort. The Lead Time Band categories and their corresponding No-Show rates remained consistent with the independently calculated results.

Because the charts are horizontal bar charts, the visual top-to-bottom display appears as **3+ weeks → 2–3 weeks → Same week**, while the underlying category sort follows the intended logical order of **Same week → 2–3 weeks → 3+ weeks**.

**Test result: PASS**

### No-Show Rate by Booking Lead Time Test

The No-Show Rate by Booking Lead Time chart was compared against the independently calculated results from the underlying dataset.

The expected No-Show Rates were:

- Same week (0–7 days): **27.8%**
- 2–3 weeks (8–21 days): **37.2%**
- 3+ weeks (22–60 days): **56.9%**

The values displayed in Power BI matched the independently calculated results.

**Test result: PASS**

### No-Show Rate by Previous No-Show History Test

The No-Show Rate by Previous No-Show History chart was compared against the independently calculated results from the underlying dataset.

The expected No-Show Rates were:

- No previous No-Show: **43.5%**
- Previous No-Show: **55.4%**

The values displayed in Power BI matched the independently calculated results.

**Test result: PASS**

### No-Show Rate by Distance to Clinic Test

The No-Show Rate by Distance to Clinic chart was compared against the independently calculated results from the underlying dataset.

The expected No-Show Rates were:

- <5 km: **46.5%**
- 5–15 km: **47.3%**
- 15 km+: **54.1%**

The values displayed in Power BI matched the independently calculated results.

**Test result: PASS**

### No-Show Rate by Reminder Status Test

The No-Show Rate by Reminder Status chart was compared against the independently calculated results from the underlying dataset.

The expected No-Show Rates were:

- No Reminder: **51.4%**
- Reminder Sent: **47.4%**

The values displayed in Power BI matched the independently calculated results.

**Test result: PASS**

### No-Show Rate by Reminder Channel Test

The No-Show Rate by Reminder Channel chart was compared against the independently calculated results from the underlying dataset.

The expected No-Show Rates were:

- SMS: **45.8%**
- Email: **48.4%**
- WhatsApp: **49.8%**

The values displayed in Power BI matched the independently calculated results.

**Test result: PASS**

### Combined Lead Time × Previous No-Show Test

The Combined Lead Time × Previous No-Show chart was compared against the independently calculated results from the underlying dataset.

The expected No-Show Rates were:

| Lead Time | No Previous No-Show | Previous No-Show |
|---|---:|---:|
| Same week | **21.8%** | **36.2%** |
| 2–3 weeks | **34.1%** | **41.9%** |
| 3+ weeks | **51.7%** | **64.1%** |

All six values displayed in Power BI matched the independently calculated results.

**Test result: PASS**

### Lead Time × Previous No-Show with Appointment Type Filter Test

The combined Lead Time × Previous No-Show chart was tested using the **Appointment Type** slicer.

The **General Consultation** appointment type was selected and the displayed No-Show Rates were compared with the independently calculated results:

| Lead Time | No Previous No-Show | Previous No-Show |
|---|---:|---:|
| Same week | **18.6%** | **33.0%** |
| 2–3 weeks | **33.4%** | **44.8%** |
| 3+ weeks | **47.7%** | **63.3%** |

The Power BI values matched the independently calculated results.

**Test result: PASS**

### Lead Time × Previous No-Show with Gender Filter Test

The combined Lead Time × Previous No-Show chart was tested using the **Gender** slicer.

The **Female** gender was selected and the displayed No-Show Rates were compared with the independently calculated results:

| Lead Time | No Previous No-Show | Previous No-Show |
|---|---:|---:|
| Same week | **24.14%** | **35.25%** |
| 2–3 weeks | **32.30%** | **42.56%** |
| 3+ weeks | **51.00%** | **65.61%** |

The Power BI values matched the independently calculated results.

**Test result: PASS**

### Lead Time × Previous No-Show with Gender Filter Test — Male

The combined Lead Time × Previous No-Show chart was tested using the **Gender** slicer.

The **Male** gender was selected and the displayed No-Show Rates were compared with the independently calculated results:

| Lead Time | No Previous No-Show | Previous No-Show |
|---|---:|---:|
| Same week | **18.75%** | **37.32%** |
| 2–3 weeks | **35.77%** | **41.78%** |
| 3+ weeks | **52.90%** | **62.52%** |

The Power BI values matched the independently calculated results.

**Test result: PASS**

In [12]:
gender_lead_previous = (
    df.groupby(["gender", "lead_time_band", "previous_no_show_group"])["appointment_outcome"]
    .apply(lambda x: (x == "No-Show").mean() * 100)
    .reset_index(name="no_show_rate")
)

gender_lead_previous

C:\Users\DELL\AppData\Local\Temp\ipykernel_8272\3336289458.py:2: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  df.groupby(["gender", "lead_time_band", "previous_no_show_group"])["appointment_outcome"]


,gender,lead_time_band,previous_no_show_group,no_show_rate
0,Female,Same week (0-7 days),No Previous No-Show,24.137931
1,Female,Same week (0-7 days),Previous No-Show,35.245902
2,Female,2-3 weeks (8-21 days),No Previous No-Show,32.303371
3,Female,2-3 weeks (8-21 days),Previous No-Show,42.561983
4,Female,3+ weeks (22-60 days),No Previous No-Show,50.997783
5,Female,3+ weeks (22-60 days),Previous No-Show,65.610860
6,Male,Same week (0-7 days),No Previous No-Show,18.750000
7,Male,Same week (0-7 days),Previous No-Show,37.323944
8,Male,2-3 weeks (8-21 days),No Previous No-Show,35.774648
9,Male,2-3 weeks (8-21 days),Previous No-Show,41.777778


### Lead Time × Previous No-Show with Gender Filter Test

The combined Lead Time × Previous No-Show chart was tested using the **Gender** slicer.

The **Female**, **Male**, and **Prefer not to say** gender selections were tested. Power BI displayed values were compared with independent calculations from the dataset.

#### Female

| Lead Time | No Previous No-Show | Previous No-Show |
|---|---:|---:|
| Same week | **24.14%** | **35.25%** |
| 2–3 weeks | **32.30%** | **42.56%** |
| 3+ weeks | **51.00%** | **65.61%** |

**Test result: PASS**

#### Male

| Lead Time | No Previous No-Show | Previous No-Show |
|---|---:|---:|
| Same week | **18.75%** | **37.32%** |
| 2–3 weeks | **35.77%** | **41.78%** |
| 3+ weeks | **52.90%** | **62.52%** |

**Test result: PASS**

#### Prefer not to say

| Lead Time | No Previous No-Show | Previous No-Show |
|---|---:|---:|
| Same week | **22.22%** | **25.00%** |
| 2–3 weeks | **37.50%** | **30.00%** |
| 3+ weeks | **41.46%** | **64.29%** |

**Note:** The "Prefer not to say" gender group showed a reversed pattern in the 2–3 weeks band (No Previous No-Show: 37.5% vs. Previous No-Show: 30.0%). Given this group's small overall size (108 records, further split across six lead-time/history combinations), this result should be interpreted cautiously as a potential small-sample anomaly rather than a genuine reversal. This is consistent with the similar finding identified for Diagnostic Test appointments in Week 6.

**Test result: PASS**

The Power BI values matched the independent Python calculations for all tested gender selections.

## Issues Identified During Testing

Testing did not identify any confirmed calculation errors in the validated KPI cards or comparison visuals. The dashboard values tested so far matched independent calculations from the underlying dataset.

However, several analytical and presentation issues requiring attention were identified.

### 1. Small-Sample Anomaly in the "Prefer not to say" Gender Group

The "Prefer not to say" gender group showed a reversed pattern in the 2–3 weeks lead-time band:

- No Previous No-Show: **37.5%**
- Previous No-Show: **30.0%**

The overall group contains only **108 records**, which are further divided across six Lead Time × Previous No-Show combinations. Therefore, this result should be interpreted cautiously as a potential small-sample anomaly rather than a genuine reversal.

This is consistent with the similar caution identified for Diagnostic Test appointments during Week 6.

**Action:** Retain the result because the dashboard calculation is correct, but avoid presenting this subgroup pattern as a general finding.

### 2. Difference in Combined Segment Rate Between Data Analytics and Data Science

A difference was identified between the highest observed Lead Time × Previous No-Show segment reported by Data Analytics (**64.1%**) and the rate reported in the Data Science analysis (**70.5%**).

The difference was attributed to a difference in denominator definition, with the Data Analytics calculation including all appointment outcomes while the Data Science calculation excluded Cancelled appointments.

**Action:** Maintain a clearly defined denominator when reporting the segment rate and document the methodological difference to prevent confusion across track outputs.

### 3. Lead Time Band Presentation

The underlying Lead Time Band order was validated as:

1. Same week (0–7 days)
2. 2–3 weeks (8–21 days)
3. 3+ weeks (22–60 days)

However, because the comparison visual is a horizontal bar chart, the visual display can appear top-to-bottom in the reverse direction.

**Action:** Confirm that the intended chronological sort is preserved and improve the visual presentation or documentation if necessary.

---

## Step 8: Dashboard Refinement

Following the Week 7 dashboard testing and issue identification, one dashboard refinement was identified and applied.

### Refinement Applied

**Lead Time Band sort order corrected**

The Week 6 dashboard displayed the Lead Time Band categories in a non-chronological order due to an unresolved circular dependency error in the sort-by-column relationship.

During Week 7 testing, this was investigated and resolved by creating a new sort column (`Lead Time Band Sort`) derived directly from `booking_lead_days`, rather than from the `Lead Time Band` text column itself. This removed the circular reference.

The affected visuals — **No-Show Rate by Booking Lead Time** and the combined **Lead Time × Previous No-Show** chart — now display categories in the correct chronological order:

1. Same week (0–7 days)
2. 2–3 weeks (8–21 days)
3. 3+ weeks (22–60 days)

### Refinement Outcome

This is the only change made to the dashboard itself during Week 7. No KPI values, chart calculations, or filtering behaviour were altered. The dashboard retains its structure of **5 KPI cards, 6 comparison visuals, and 2 slicers**.

---

## Step 9: Re-test After Refinement

After applying the dashboard refinement, the affected and previously validated components were re-tested to confirm that the changes did not introduce new errors or alter the analytical results.

### Re-testing Performed

The following components were re-tested:

- **5 KPI cards** — values remained correct.
- **6 comparison visuals** — displayed values remained consistent with the validated results.
- **Appointment Type slicer** — filtering continued to work correctly.
- **Gender slicer** — filtering continued to work correctly.
- **Lead Time × Previous No-Show chart** — calculations and filtering remained correct.
- **Lead Time Band ordering** — categories remained in the intended chronological order:
  1. Same week (0–7 days)
  2. 2–3 weeks (8–21 days)
  3. 3+ weeks (22–60 days)

### Re-test Result

All tested dashboard components continued to function correctly after the refinement.

No new calculation errors, filtering issues, or unexpected visual behaviour were identified.

**Overall re-test result: PASS**

The refined dashboard is therefore ready for the next stage of Week 7 validation and cross-track testing.

---

## Step 10: Cross-Track Testing Contribution

In [13]:
donald_segment = df[
    (df["booking_lead_days"] > 30) &
    (df["previous_no_shows"] >= 1)
]

# Excluding Cancelled (matching Donald's original denominator)
donald_no_cancelled = donald_segment[donald_segment["appointment_outcome"] != "Cancelled"]
donald_rate_excl_cancelled = (donald_no_cancelled["appointment_outcome"] == "No-Show").mean() * 100

# Including Cancelled (matching Donald's retest)
donald_rate_incl_cancelled = (donald_segment["appointment_outcome"] == "No-Show").mean() * 100

print(f"Donald's segment (>30 days), excluding Cancelled: {donald_rate_excl_cancelled:.2f}%")
print(f"Donald's segment (>30 days), including Cancelled: {donald_rate_incl_cancelled:.2f}%")
print(f"Sample size: {len(donald_segment)} total appointments")

Donald's segment (>30 days), excluding Cancelled: 70.52%
Donald's segment (>30 days), including Cancelled: 67.88%
Sample size: 1015 total appointments


## Independent Replication of Data Science Segment Definition

To validate the Data Science result independently, I reproduced Donald's exact segment definition using the Data Analytics dataset.

The segment was defined as:

- **Booking lead time:** `booking_lead_days > 30`
- **Previous No-Show history:** `previous_no_shows ≥ 1`

I calculated the No-Show Rate using two denominator definitions:

1. **Cancelled appointments excluded**, matching Donald's original calculation.
2. **Cancelled appointments included**, matching Donald's retest.

### Results

| Denominator Definition | No-Show Rate |
|---|---:|
| Cancelled excluded | **70.52%** |
| Cancelled included | **67.88%** |

The segment contained **1,015 appointments** in total.

The independently calculated results matched Donald's reported figures exactly:

- **70.52%** when Cancelled appointments were excluded.
- **67.88%** when Cancelled appointments were included.

This confirms that the Data Analytics analysis can reproduce the Data Science calculation using the same segment definition and underlying dataset.

## HC-POD Cross-Track Testing: Segment Rate Discrepancy Validation

**Track collaborated with:** Data Science (Donald Nwachukwu)

**Component/output being tested:** The Lead Time × Previous No-Show highest-observed segment rate, previously reported differently by the two tracks (64.1% Data Analytics vs. 70.5% Data Science).

**Information/output received:** Donald's exact segment definition (`booking_lead_days > 30` AND `previous_no_shows ≥ 1`) and denominator definitions (Cancelled excluded: 689 ÷ 977 = 70.52%; Cancelled included: 689 ÷ 1015 = 67.88%).

**Information/output provided:** A request to re-run his calculation with Cancelled appointments included, followed by an independent replication of his exact segment definition on the Data Analytics dataset.

**Testing activity completed:**
1. Donald re-ran his segment with Cancelled appointments included, producing 67.88%.
2. The Data Analytics track independently recalculated Donald's exact segment definition (`>30 days`, `≥1 previous no-show`) on its own copy of the dataset, testing both denominator variants.

**Issue/finding identified:** The independent recalculation exactly matched Donald's figures (70.52% excluding Cancelled, 67.88% including Cancelled), confirming that both tracks are working from the same underlying data. The testing also identified two definitional differences between the original analyses: (1) the long lead-time threshold (Data Analytics: `≥22 days`; Data Science: `>30 days`), and (2) the denominator definition (Data Analytics includes Cancelled appointments; Data Science originally excluded them).

**Action taken:** Both tracks' exact segment definitions were documented explicitly, including the lead-time threshold, previous No-Show condition, and denominator definition, so future comparisons state these criteria alongside any reported rate.

**Retest result:** Full convergence was confirmed for the Data Science segment definition. Recalculating Donald's exact criteria on the Data Analytics dataset reproduced his figures precisely: **70.52% excluding Cancelled** and **67.88% including Cancelled**. This validates the Data Science calculation and confirms that the underlying dataset is consistent across the two analyses.

**What changed as a result:** The investigation moved from the Week 6 explanation that the discrepancy was primarily a denominator issue to a more precise understanding that the two analyses used different segment definitions, including different lead-time thresholds and denominator definitions. This demonstrates the importance of explicitly documenting segment criteria when comparing analytical rates across tracks.

**Evidence:** Donald's message containing the exact segment definitions and calculations, together with the Data Analytics notebook cell independently reproducing the results (**70.52%**, **67.88%**, **n=1015**).

---

## Step 11: Validated Key Findings

Week 7 testing was used to confirm whether the important analytical findings from Week 6 remained accurate when independently recalculated and tested across dashboard filters, visuals, and segment definitions.

### 1. Overall No-Show Rate

The overall No-Show Rate was independently validated at **48.5%**.

The value displayed in the Power BI KPI card matched the independent calculation from the underlying dataset.

**Validation result: Confirmed**



### 2. Booking Lead Time

Booking lead time remains an important observed factor associated with No-Show rates.

The independently validated rates were:

| Lead Time | No-Show Rate |
|---|---:|
| Same week (0–7 days) | **27.8%** |
| 2–3 weeks (8–21 days) | **37.2%** |
| 3+ weeks (22–60 days) | **56.9%** |

The pattern was also validated across appointment types and distance groups during Week 6, and the Power BI visual values matched the independent calculations during Week 7 testing.

Statistical validation from Week 6 also showed an association between booking lead time and appointment outcome (**χ² = 280.09, p < 0.0001, Cramér's V = 0.167**).

**Validation result: Confirmed**



### 3. Previous No-Show History

Appointments with previous No-Show history had a higher observed No-Show Rate than appointments without previous No-Show history:

| Previous No-Show History | No-Show Rate |
|---|---:|
| No previous No-Show | **43.5%** |
| Previous No-Show | **55.4%** |

The difference of **11.9 percentage points** was reproduced during dashboard validation.

Week 6 statistical testing also identified an association between previous No-Show history and appointment outcome (**χ² = 69.86, p < 0.0001, Cramér's V = 0.118**).

**Validation result: Confirmed**



### 4. Combined Lead Time × Previous No-Show Pattern

The combined analysis remains an important observed segmentation pattern.

The independently validated No-Show Rates were:

| Lead Time | No Previous No-Show | Previous No-Show |
|---|---:|---:|
| Same week | **21.8%** | **36.2%** |
| 2–3 weeks | **34.1%** | **41.9%** |
| 3+ weeks | **51.7%** | **64.1%** |

The highest observed rate in the Data Analytics segmentation was **64.1%**, occurring in the **3+ weeks + previous No-Show** segment.

The combined six-category analysis was statistically validated in Week 6 (**χ² = 358.52, p < 0.0001, Cramér's V = 0.189**), which was the strongest association among the tested segmentation approaches.

The Power BI combined visual was also independently tested using Appointment Type and Gender filters, with the displayed values matching the independently calculated results.

**Validation result: Confirmed, with segment-definition clarification documented separately.**



### 5. Dashboard KPI and Visual Accuracy

The five KPI cards and six comparison visuals were independently checked against calculations from the underlying dataset.

The following dashboard components passed validation:

- Overall No-Show Rate
- No Reminder
- Previous No-Show
- 3+ Week Lead Time
- Highest Observed Segment
- Booking Lead Time visual
- Previous No-Show visual
- Distance visual
- Reminder Status visual
- Reminder Channel visual
- Combined Lead Time × Previous No-Show visual

The Appointment Type and Gender slicers were also tested and responded as expected.

**Validation result: Confirmed**



### 6. Cross-Track Segment Discrepancy

The previously reported **64.1% vs. 70.5%** discrepancy between Data Analytics and Data Science was investigated through cross-track testing.

The Data Analytics track independently reproduced Donald's exact Data Science segment definition:

- `booking_lead_days > 30`
- `previous_no_shows ≥ 1`

The resulting rates matched Donald's calculations exactly:

- **70.52%** when Cancelled appointments were excluded.
- **67.88%** when Cancelled appointments were included.

The testing identified differences in the definitions used by the two analyses, particularly the **lead-time threshold** and **denominator definition**.

Therefore, the two reported rates should not be treated as directly comparable unless the segment definition and denominator are explicitly stated.

**Validation result: Confirmed and clarified**



### Overall Validation Conclusion

Week 7 testing confirmed that the major Data Analytics findings carried forward from Week 6 were reproducible and that the Power BI dashboard values were consistent with independent calculations.

The testing also clarified the importance of explicitly defining segment thresholds and denominator rules when comparing analytical results across tracks.

The validated findings therefore provide a stronger basis for the updated business insights and recommendations in the following sections.

---

## Step 12: Updated Business Insights

The Week 7 testing confirmed that the main analytical findings from Week 6 were reproducible and that the Power BI dashboard values were consistent with independent calculations. Based on these validated results, the following business insights were refined.

### 1. Longer Booking Lead Times Are Associated With Higher No-Show Rates

The validated analysis shows a clear increase in observed No-Show Rate as booking lead time increases:

- **Same week (0–7 days): 27.8%**
- **2–3 weeks (8–21 days): 37.2%**
- **3+ weeks (22–60 days): 56.9%**

The pattern was also observed across different appointment types and distance groups.

**Business insight:** Appointments scheduled further in advance may require greater attention to appointment follow-up and confirmation because they showed higher observed No-Show Rates in this dataset.



### 2. Previous No-Show History Remains an Important Segmenting Factor

Appointments with previous No-Show history had a validated No-Show Rate of **55.4%**, compared with **43.5%** for appointments without previous No-Show history.

This difference of **11.9 percentage points** was consistent with the Week 6 analysis and remained present during Week 7 validation.

**Business insight:** Previous attendance behaviour can be used as a practical segmentation factor when considering which appointment groups may require additional attention.



### 3. Combining Lead Time and Previous No-Show History Provides More Detailed Segmentation

The combined analysis provides more detail than considering either factor separately.

The highest observed Data Analytics segment was:

**3+ weeks lead time + previous No-Show history → 64.1% No-Show Rate**

compared with:

**Same week + no previous No-Show history → 21.8%**

The combined six-category analysis also produced the highest Cramér's V among the tested analytical associations (**V = 0.189**).

**Note:** The **64.1%** figure is specific to the Data Analytics segment definition (`booking_lead_days ≥ 22 days`, denominator including Cancelled appointments). A parallel Data Science analysis using a different lead-time threshold (`>30 days`) and a different denominator definition originally reported **70.5%**. Cross-track testing reproduced the Data Science figures under its stated definitions and confirmed that the two analyses used different segment and denominator definitions. This highlights the importance of stating segment criteria explicitly when reporting or comparing No-Show Rate figures.

**Business insight:** Combining booking lead time with previous No-Show history can help HealthConnect distinguish between appointment groups with substantially different observed No-Show Rates rather than treating all appointments as having the same level of observed No-Show behaviour.



### 4. Reminder Status Shows a Difference, but Does Not Establish Reminder Effectiveness

The validated dashboard results showed:

- **No Reminder:** 51.4% No-Show Rate
- **Reminder Sent:** 47.4% No-Show Rate

The difference is **4.0 percentage points**.

However, this analysis is observational and does not establish that sending a reminder caused the lower No-Show Rate.

Reminder channel results also showed relatively small differences:

- **SMS:** 45.8%
- **Email:** 48.4%
- **WhatsApp:** 49.8%

**Business insight:** Reminder status and channel provide useful descriptive information for monitoring appointment behaviour, but the current analysis does not provide sufficient evidence to conclude that one reminder approach is more effective than another.



### 5. Distance Shows a Supporting Relationship With No-Show Behaviour

The validated distance analysis showed:

- **<5 km:** 46.5%
- **5–15 km:** 47.3%
- **15 km+:** 54.1%

The highest distance group therefore had a **7.6 percentage-point** higher observed No-Show Rate than the <5 km group.

However, the Week 6 Data Science modelling showed that distance did not provide additional predictive value in the reported model setup.

**Business insight:** Distance remains useful as a supporting descriptive factor, but it should not be treated as a primary predictive factor based on the current evidence.



### 6. Appointment Type Differences Are Relatively Narrow

The validated appointment-type analysis showed:

- **Diagnostic Test:** 49.7%
- **Follow-up:** 51.2%
- **General Consultation:** 46.6%
- **Specialist Consultation:** 47.4%

The difference between the highest and lowest observed rates was **4.6 percentage points**.

**Business insight:** Appointment type provides useful context for understanding attendance patterns, but the observed differences were relatively narrow compared with the differences associated with booking lead time and previous No-Show history.



### Overall Updated Business Insight

The Week 7 validation strengthens the evidence for focusing operational attention on **booking lead time and previous No-Show history**, particularly when the two factors are considered together.

At the same time, the testing showed that other variables such as reminders, distance, and appointment type should be interpreted according to the strength and limitations of the evidence rather than being treated as confirmed causes of No-Shows.

The main business value of the analysis is therefore the ability to identify and monitor appointment segments with different observed No-Show patterns, while keeping the distinction between **descriptive association and predictive or causal conclusions** clear.

---

## Step 13: Updated Recommendations

Based on the findings validated during Week 7, the following recommendations were refined to support HealthConnect's appointment management and No-Show reduction efforts.

### 1. Prioritise Appointments With Longer Booking Lead Times

HealthConnect should pay closer attention to appointments scheduled further in advance, particularly those in the **3+ weeks** lead-time group, which had a validated No-Show Rate of **56.9%** compared with **27.8%** for same-week appointments.

**Recommended action:**
- Monitor appointments with longer booking lead times.
- Consider additional confirmation or follow-up closer to the appointment date.
- Track whether attendance patterns change after any operational intervention is introduced.

This recommendation is based on an observed association and should be evaluated using future operational data before concluding that the intervention directly reduces No-Shows.



### 2. Give Additional Attention to Patients With Previous No-Show History

Appointments involving patients with previous No-Show history had a validated No-Show Rate of **55.4%**, compared with **43.5%** for appointments without previous No-Show history.

**Recommended action:**
- Use previous No-Show history as a segmentation factor for appointment monitoring.
- Consider additional confirmation or follow-up for appointments belonging to this group.
- Monitor subsequent attendance outcomes to evaluate whether the approach is useful.



### 3. Combine Lead Time and Previous No-Show History for Appointment Segmentation

The combined analysis identified substantially different observed No-Show Rates across the six Lead Time × Previous No-Show segments. The highest observed Data Analytics segment was **3+ weeks + previous No-Show history at 64.1%**.

**Recommended action:**
- Consider combining booking lead time and previous No-Show history when prioritising appointments for monitoring.
- Use clearly documented segment definitions whenever rates are reported.
- If this segmentation is used in future predictive modelling, maintain the exact definitions of the lead-time threshold and previous No-Show condition.

The **64.1%** figure should be interpreted specifically according to the Data Analytics definition (`booking_lead_days ≥ 22 days` with Cancelled appointments included in the denominator). The Data Science analysis used a different threshold and denominator definition.



### 4. Continue Monitoring Reminder Activity Without Assuming Causality

The validated analysis showed a **4.0 percentage-point** difference between appointments with no reminder (**51.4%**) and appointments where a reminder was sent (**47.4%**).

However, the current analysis does not establish that reminders caused the difference.

**Recommended action:**
- Continue monitoring reminder status and channel alongside attendance outcomes.
- If HealthConnect wants to assess reminder effectiveness, consider a controlled or well-designed evaluation using future operational data.
- Avoid selecting a reminder channel as the preferred approach based solely on the current descriptive differences.



### 5. Treat Distance as a Supporting Factor

The 15 km+ group had a higher observed No-Show Rate (**54.1%**) than the <5 km group (**46.5%**). However, Data Science modelling indicated that distance did not provide additional predictive value in the reported model setup.

**Recommended action:**
- Retain distance as a supporting monitoring variable.
- Avoid making distance a primary basis for intervention or prioritisation without further evidence.
- Reassess its usefulness when additional data or modelling approaches become available.



### 6. Use the Validated Dashboard for Ongoing Monitoring

The Week 7 testing confirmed the accuracy of the five KPI cards, six comparison visuals, and two dashboard slicers.

**Recommended action:**
- Use the Power BI dashboard to monitor No-Show patterns across key segments.
- Maintain the validated KPI definitions and documented segment criteria.
- Recheck dashboard calculations whenever the dataset structure, DAX measures, or segment definitions are changed.



### Overall Recommendation

The strongest operational focus from the current analysis is on **booking lead time and previous No-Show history**, particularly when the two factors are considered together.

HealthConnect should use these findings to support **targeted monitoring and follow-up**, while treating the current results as evidence of observed relationships rather than proof of causation.

Future operational or predictive analysis should also maintain clear definitions for **segment thresholds, denominator rules, and outcome categories** so that results remain comparable across teams and over time.

---

## Step 14: Before/After Improvements

Week 7 testing resulted in improvements to both the Power BI dashboard presentation and the interpretation of the analytical results.

### 1. Lead Time Band Ordering

**Before:**  
The Lead Time Band categories had an unresolved sorting issue in Power BI. The intended chronological order was:

1. Same week (0–7 days)
2. 2–3 weeks (8–21 days)
3. 3+ weeks (22–60 days)

The issue was related to the sort-by-column setup and had previously resulted in a circular dependency error.

**After:**  
A separate `Lead Time Band Sort` column was created directly from `booking_lead_days` and used to control the category ordering.

The Lead Time Band categories now follow the intended chronological order.

**Improvement:**  
The dashboard presentation is clearer and the lead-time categories can be interpreted in the correct sequence.



### 2. Segment Definition Documentation

**Before:**  
The Week 6 comparison between the Data Analytics and Data Science results reported **64.1% vs. 70.5%**, with the difference initially attributed mainly to denominator definitions.

**After:**  
Week 7 cross-track testing independently reproduced the Data Science results and identified differences in both the **lead-time threshold** and **denominator definition** used by the two analyses.

The Data Analytics definition used:

- `booking_lead_days ≥ 22`
- `previous_no_shows ≥ 1`
- Cancelled appointments included in the denominator

The Data Science definition used:

- `booking_lead_days > 30`
- `previous_no_shows ≥ 1`
- Cancelled appointments originally excluded from the denominator

**Improvement:**  
Segment thresholds and denominator definitions are now explicitly documented whenever the No-Show Rate is reported or compared across analyses.



### 3. Cross-Track Validation of Reported Results

**Before:**  
The difference between the two reported segment rates had not been independently reproduced or fully validated.

**After:**  
The Data Analytics track independently applied the Data Science segment definition to the HealthConnect dataset and reproduced:

- **70.52%** with Cancelled appointments excluded.
- **67.88%** with Cancelled appointments included.

**Improvement:**  
The cross-track testing confirmed that the Data Science calculation could be independently reproduced using the same underlying dataset and clarified the methodological differences between the two analyses.



### Overall Before/After Improvement

| Area | Before Week 7 | After Week 7 |
|---|---|---|
| Lead Time Band ordering | Sort-order issue remained | Chronological ordering corrected |
| Segment definitions | Not fully explicit when comparing rates | Thresholds and denominator definitions documented |
| 64.1% vs. 70.5% discrepancy | Explanation was incomplete | Cross-track testing identified the differing definitions |
| Data Science calculation | Not independently reproduced | 70.52% and 67.88% independently reproduced |
| Dashboard validation | Testing in progress | Core KPIs, visuals, slicers, and refined ordering re-tested successfully |

### Overall Improvement

Week 7 changed the HealthConnect analysis from a dashboard and set of findings that had been developed in Week 6 into a more **tested, validated, and clearly documented analytical output**. The main improvements were the correction of the Lead Time Band ordering and the clarification of segment definitions used when comparing results across tracks.

---

## Step 15: Limitations

Although the Week 7 testing confirmed the accuracy of the main dashboard calculations and analytical findings, several limitations remain relevant when interpreting the HealthConnect analysis.

### 1. Synthetic Dataset

The HealthConnect dataset is a synthetic dataset. Therefore, the observed No-Show patterns should not automatically be treated as representative of real-world patient behaviour or actual HealthConnect operations.

The overall No-Show Rate of **48.5%**, for example, should be interpreted as a characteristic of this dataset rather than as a real-world benchmark.



### 2. Previous Appointment History May Not Represent Real Patient Histories

The `previous_appointments` and `previous_no_shows` fields may not represent fully coherent longitudinal patient histories.

Although the Week 4 validation confirmed that `previous_no_shows` does not exceed `previous_appointments`, the dataset structure does not allow the analysis to independently verify the complete sequence of each patient's historical appointments.

Therefore, findings involving previous No-Show history should be interpreted within the limits of the available dataset.



### 3. Missing Values Remain in Some Variables

Some variables contain missing values:

- `reminder_channel`: **1,366 records (27.3%)**
- `distance_to_clinic_km`: **90 records (1.8%)**
- `waiting_time_minutes`: **60 records (1.2%)**

The missingness in `reminder_channel` is structurally related to appointments where no reminder was sent.

The relatively small amount of missingness in distance and waiting time should still be considered when interpreting analyses involving those variables.



### 4. Observational Analysis Does Not Establish Causation

The analysis identifies associations and observed differences between groups. It does not establish that a particular factor directly causes a No-Show.

For example, the lower observed No-Show Rate among appointments where reminders were sent does not by itself prove that reminders caused the difference.

Further controlled analysis or operational testing would be required to establish causal effects.



### 5. Statistical Association Does Not Equal Predictive Performance

The Week 6 statistical tests confirmed associations between appointment outcome and factors such as booking lead time and previous No-Show history.

However, statistical association does not by itself demonstrate that a variable will produce strong predictions for individual appointments.

The Data Science modelling results provide additional predictive context, but model performance remains dependent on the available features, dataset, modelling approach, and validation methodology.



### 6. Segment Definitions Affect Reported Rates

Week 7 cross-track testing demonstrated that No-Show Rates can change when the segment definition changes.

The original **64.1%** Data Analytics result used a lead-time threshold of `booking_lead_days ≥ 22` and included Cancelled appointments in the denominator.

The Data Science analysis used `booking_lead_days > 30` and originally excluded Cancelled appointments.

The testing reproduced the Data Science results under its stated definitions (**70.52% excluding Cancelled and 67.88% including Cancelled**).

Therefore, No-Show Rates from different analyses should only be compared when the segment criteria and denominator definitions are clearly aligned.



### 7. Small Subgroups Require Cautious Interpretation

Some categories contain substantially fewer observations than others.

For example, the **Prefer not to say** gender group contains only **108 records**, which are further divided across multiple lead-time and previous No-Show combinations.

The reversed pattern observed in the 2–3 week segment should therefore be interpreted cautiously and should not be treated as a general behavioural pattern without additional data.



### 8. Dashboard Validation Covers the Tested Configuration

The Week 7 testing confirmed the five KPI cards, six comparison visuals, slicers, combined analysis, and Lead Time Band ordering under the tested conditions.

However, dashboard accuracy can change if the underlying dataset, DAX measures, relationships, filters, or segment definitions are modified in the future.

Any substantial changes to the dashboard should therefore be followed by another validation cycle.



### Overall Limitation

The Week 7 testing increased confidence in the accuracy and consistency of the current analysis, but the findings remain limited by the synthetic dataset, available variables, missing values, observational design, subgroup sizes, and differences in analytical definitions.

These limitations should be considered when interpreting the findings and when applying the analysis to future HealthConnect operational or predictive work.

---

## Step 16: Remaining Analytical Issues

Week 7 testing resolved the identified dashboard sorting issue and clarified the discrepancy between the Data Analytics and Data Science segment calculations. No confirmed calculation errors remain in the tested dashboard components.

However, the following analytical issues and considerations remain relevant for future HealthConnect analysis.

### 1. Segment Definition Alignment Across Tracks

The cross-track testing confirmed that the different reported segment rates resulted from differences in the analytical definitions used by the two tracks.

The Data Analytics analysis used:

- `booking_lead_days ≥ 22`
- `previous_no_shows ≥ 1`
- Cancelled appointments included in the denominator

The Data Science analysis used:

- `booking_lead_days > 30`
- `previous_no_shows ≥ 1`
- Cancelled appointments originally excluded from the denominator

Although the Data Science calculation was independently reproduced, the two rates should not be directly compared unless the segment definition and denominator are aligned.

**Remaining consideration:** Future cross-track analyses should agree on and document segment definitions before comparing rates.



### 2. Causal Impact of Reminders Remains Unresolved

The analysis identified a difference in No-Show Rates between appointments with and without reminders, but the current dataset does not establish whether reminders caused the observed difference.

The effectiveness of reminder channels also remains uncertain because the observed differences between SMS, Email, and WhatsApp were relatively small.

**Remaining consideration:** A future controlled or well-designed operational analysis would be required to evaluate the causal effect of reminders.



### 3. Predictive Value of Individual Factors Requires Continued Validation

Booking lead time and previous No-Show history showed statistically validated associations with appointment outcomes and were also useful in the Data Science modelling work.

However, the current analysis does not establish that these variables alone provide sufficient predictive performance for individual appointments.

**Remaining consideration:** Future modelling should continue evaluating feature combinations, model performance, and generalisation using appropriate validation methods.



### 4. Dataset Limitations Affect Further Interpretation

The synthetic nature of the dataset, missing values, and limitations in the available patient-history information remain constraints.

These limitations mean that the current findings should not be assumed to represent actual HealthConnect patient behaviour without validation against real operational data.

**Remaining consideration:** Future analysis using real or more representative operational data should reassess whether the observed patterns remain consistent.

    

### 5. Some Subgroup Patterns Require More Data

The analysis identified some unusual patterns in smaller subgroups, including the **Prefer not to say** gender group and the Diagnostic Test 2–3 week segment.

These patterns were retained as observations but were not treated as general conclusions because of the smaller subgroup sizes.

**Remaining consideration:** These patterns should be reassessed if additional data becomes available.



### Overall Status

At the end of Week 7:

- **No confirmed KPI calculation errors remain in the tested dashboard.**
- **Dashboard filtering and visual calculations passed testing.**
- **Lead Time Band ordering was corrected and successfully re-tested.**
- **The Data Science segment calculation was independently reproduced.**
- **The 64.1% vs. 70.5% discrepancy was clarified as a difference in analytical definitions.**
- **Causal effects of reminders remain unestablished.**
- **The generalisability of findings beyond the synthetic dataset remains limited.**

These remaining issues do not prevent the current analytical output from being used as a validated project deliverable, but they should be considered in future HealthConnect analysis and development.

---

## Step 17: Week 8 Recommendations

Based on the Week 7 testing, validation, and refinement activities, the following recommendations are proposed for the next stage of the HealthConnect project.

### 1. Build on the Validated Lead Time and Previous No-Show Findings

Week 7 confirmed the consistency of the observed relationship between booking lead time, previous No-Show history, and appointment outcomes.

**Recommendation:**
- Carry the validated Lead Time × Previous No-Show segmentation forward into Week 8.
- Maintain the documented segment definitions when using or comparing these results.
- Continue evaluating whether these factors remain useful as the HealthConnect solution develops.



### 2. Maintain Clear Analytical Definitions Across Tracks

The Week 7 cross-track testing showed that differences in segment thresholds and denominator definitions can produce different No-Show Rates from the same underlying dataset.

**Recommendation:**
- Define analytical criteria before cross-track comparisons are performed.
- Document lead-time thresholds, No-Show conditions, and denominator rules alongside reported metrics.
- Use consistent definitions when different tracks need to compare or integrate analytical outputs.



### 3. Continue Cross-Track Integration and Validation

The Data Analytics and Data Science collaboration helped validate the reported segment calculations and identify differences in analytical definitions.

**Recommendation:**
- Continue sharing validated findings and clearly defined analytical outputs with relevant HealthConnect tracks.
- Where another track uses an analytical result from Data Analytics, provide the calculation definition and denominator alongside the result.
- Use cross-track testing when an output becomes a dependency for another part of the HealthConnect solution.



### 4. Further Evaluate Predictive Use of the Validated Segments

The combined Lead Time × Previous No-Show analysis showed the strongest observed association among the tested segmentation approaches, while Data Science modelling provided additional predictive context.

**Recommendation:**
- Continue evaluating the combined segment in future predictive analysis.
- Compare its usefulness with individual features such as booking lead time and previous No-Show history.
- Avoid treating the segment as a confirmed predictive or causal factor without further model validation.



### 5. Evaluate Reminder Effectiveness With Future Operational Data

The current analysis identified differences between reminder groups and channels but did not establish causal effectiveness.

**Recommendation:**
- If operational data becomes available, evaluate reminder effectiveness using a suitable testing or evaluation approach.
- Continue monitoring reminder status and channel alongside appointment outcomes.
- Avoid making operational decisions about the most effective reminder channel based solely on the current dataset.



### 6. Reassess Findings Using More Representative Data

The current HealthConnect dataset is synthetic, which limits how confidently the observed patterns can be applied to real-world operations.

**Recommendation:**
- If real operational data becomes available, reassess the validated Week 7 findings.
- Compare whether the relationships involving lead time, previous No-Show history, distance, and reminders remain consistent.
- Recalculate KPIs using the same documented definitions to maintain comparability.



### 7. Preserve the Validated Dashboard as a Baseline

The Week 7 testing confirmed the main KPI cards, comparison visuals, slicers, calculations, and Lead Time Band ordering.

**Recommendation:**
- Treat the validated Week 7 dashboard as a baseline for future development.
- Re-test KPI calculations and filters whenever major changes are made.
- Preserve the current KPI and segment definitions unless a documented analytical change is introduced.



### Overall Week 8 Direction

Week 8 should build on the validated Week 7 outputs rather than restarting the analysis.

The main priorities should be to:

1. **Carry forward the validated analytical findings.**
2. **Maintain consistent definitions across HealthConnect tracks.**
3. **Continue cross-track integration where analytical outputs become dependencies.**
4. **Further evaluate the predictive usefulness of the combined segmentation.**
5. **Reassess the findings when more representative data becomes available.**
6. **Use the validated Power BI dashboard as the baseline for future improvements.**

These recommendations provide a transition from Week 7 testing and validation toward continued HealthConnect integration and development in Week 8.